### PR TITLE
197 entity spec

### LINKED ISSUE #197: Allow for "entity specs"
I would love to see [entity specs](https://docs.datomic.com/on-prem/schema.html#entity-specs), especially [required attributes](https://docs.datomic.com/on-prem/schema.html#required-attributes) implemented in datahike.

### PR DESCRIPTION
**Pull request by kordano**
Closes #197

### CODE REVIEW

File: test/datahike/test/entity_spec.cljc [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    (letfn [(tx-with-ensure [account]
+              (d/transact conn [(assoc account :db/ensure :account/guard)]))]
+      (testing "assert valid account"
+        (let [{:keys [db-after]} (tx-with-ensure valid-account)]
+          (is (= valid-account (d/pull db-after '[:account/email :account/balance] [:account/email (:account/email valid-account)])))))
+      (testing "assert invalid account with one invalid predicate"
+        (is (thrown-msg?
+             "Entity 5 failed predicates #{datahike.test.entity-spec/is-email?} of spec :account/guard"
+             (tx-with-ensure invalid-account))))
+      (testing "assert invalid account with mulitple invalid predicates"
```
Comment (TimoKramer) [inline test/datahike/test/entity_spec.cljc:107]: Typo mulitple -> multiple

File: test/datahike/test/entity_spec.cljc [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+          (is (= valid-account (d/pull db-after '[:account/email :account/balance] [:account/email (:account/email valid-account)])))))
+      (testing "assert invalid account with missing attributes"
+        (is (thrown-msg?
+             "Entity 5 missing attributes #{:account/balance} of spec :account/guard"
+             (tx-with-ensure invalid-account-attr))))
+      (testing "assert invalid account with one invalid predicate"
+        (is (thrown-msg?
+             "Entity 5 failed predicates #{datahike.test.entity-spec/is-email?} of spec :account/guard"
+             (tx-with-ensure invalid-account-pred))))
+      (testing "assert invalid account with mulitple invalid predicates"
```
Comment (TimoKramer) [inline test/datahike/test/entity_spec.cljc:146]: Typo mulitple -> multiple

**Review by TimoKramer — DISMISSED** [review]
found two typos and tested single entity predicate without problems. Seems to work accordingly to Datomic :+1: . Some README update required?

**Comment by kordano** [comment]
I'll fix the typos and add a small info in the README.

**Comment by kordano** [comment]
Added a doc with an example.

**Review by TimoKramer — APPROVED** [review]

**Comment by TimoKramer** [comment]
Super!

**Pull request merged**